src/Collection.php

```diff
@@ -666,13 +667,17 @@ public function pair(): CollectionInterface
 
     public function partition(callable ...$callbacks): CollectionInterface
     {
-        return new self(
-            Pipe::of()(
-                Partition::of()(...$callbacks),
-                Map::of()(static fn (array $callUserFuncParameters): CollectionInterface => self::fromCallable(...$callUserFuncParameters)),
-            ),
-            [$this->getIterator()]
-        );
+        $mapCallback = static function (array $partitionResult): CollectionInterface {
+            /**
+             * @var Closure(Iterator<TKey, T>): Generator<TKey, T> $callback
+             * @var array{0: Iterator<TKey, T>} $parameters
+             */
+            [$callback, $parameters] = $partitionResult;
+
+            return self::fromCallable($callback, $parameters);
+        };
+
+        return new self(Pipe::of()(Partition::of()(...$callbacks), Map::of()($mapCallback)), [$this->getIterator()]);
```

Comment:
What do you think about doing this:

```diff
diff --git a/src/Collection.php b/src/Collection.php
index 640df98a..8042ecd5 100644
--- a/src/Collection.php
+++ b/src/Collection.php
@@ -667,15 +667,11 @@ final class Collection implements CollectionInterface
 
     public function partition(callable ...$callbacks): CollectionInterface
     {
-        $mapCallback = static function (array $partitionResult): CollectionInterface {
+        $mapCallback =
             /**
-             * @var Closure(Iterator<TKey, T>): Generator<TKey, T> $callback
-             * @var array{0: Iterator<TKey, T>} $parameters
+             * @param array{0: Closure(Iterator<TKey, T>): Generator<TKey, T>, 1: array{0: Iterator<TKey, T>}} $partitionResult
              */
-            [$callback, $parameters] = $partitionResult;
-
-            return self::fromCallable($callback, $parameters);
-        };
+            static fn (array $partitionResult): CollectionInterface => self::fromCallable(...$partitionResult);
 
         return new self(Pipe::of()(Partition::of()(...$callbacks), Map::of()($mapCallback)), [$this->getIterator()]);
     }
```

Reply:
So we don't `unwrap` and `wrap` parameters.

Reply:
already tried that, doesn't work because PHPStan doesn't yet support reading doc blocks above closures: https://github.com/phpstan/phpstan/issues/3770

Reply:
Oh... I'll add a TODO and merge!

Reply:
in other cases it doesn't mind and just treats it as `callable` and `mixed`, but here it needs to know the return type for the `$callback` so that it can link it to the template types of `fromCallable`